gdal/ogr/ogrsf_frmts/cad/ogrcaddriver.cpp

```diff
@@ -151,30 +151,53 @@ void RegisterOGRCAD()
-        /* 6 8859_5 */  "ISO-8859-5", /* 7 8859_6 */  "ISO-8859-6",
-        /* 8 8859_7 */  "ISO-8859-7", /* 9 8859_8 */  "ISO-8859-8",
-        /* 10 8859_9 */  "ISO-8859-9", /* 11 DOS437 */  "CP437",
-        /* 12 DOS850 */  "CP850", /* 13 DOS852 */  "CP852",
-        /* 14 DOS855 */  "CP855", /* 15 DOS857 */  "CP857",
-        /* 16 DOS860 */  "CP860", /* 17 DOS861 */  "CP861",
-        /* 18 DOS863 */  "CP863", /* 19 DOS864 */  "CP864",
-        /* 20 DOS865 */  "CP865", /* 21 DOS869 */  "CP869",
-        /* 22 DOS932 */  "CP932", /* 23 MACINTOSH */ "MACINTOSH",
-        /* 24 BIG5 */  "BIG5", /* 25 KSC5601 */  "CP949",
-        /* 26 JOHAB */  "JOHAB", /* 27 DOS866 */  "CP866",
-        /* 28 ANSI_1250 */ "CP1250", /* 29 ANSI_1251 */ "CP1251",
-        /* 30 ANSI_1252 */ "CP1252", /* 31 GB2312 */  "GB2312",
-        /* 32 ANSI_1253 */ "CP1253", /* 33 ANSI_1254 */ "CP1254",
-        /* 34 ANSI_1255 */ "CP1255", /* 35 ANSI_1256 */ "CP1256",
-        /* 36 ANSI_1257 */ "CP1257", /* 37 ANSI_874 */ "CP874",
-        /* 38 ANSI_932 */ "CP932", /* 39 ANSI_936 */ "CP936",
-        /* 40 ANSI_949 */ "CP949", /* 41 ANSI_950 */ "CP950",
-        /* 42 ANSI_1361 */ "CP1361", /* 43 ANSI_1200 */ "UTF-16",
+        /* 0 UNDEFINED */ "",
+        /* 1 ASCII */ "US-ASCII",
+        /* 2 8859_1 */ "ISO-8859-1",
+        /* 3 8859_2 */ "ISO-8859-2",
+        /* 4 UNDEFINED */ "",
+        /* 5 8859_4 */ "ISO-8859-4",
+        /* 6 8859_5 */ "ISO-8859-5",
+        /* 7 8859_6 */ "ISO-8859-6",
+        /* 8 8859_7 */ "ISO-8859-7",
+        /* 9 8859_8 */ "ISO-8859-8",
+        /* 10 8859_9 */ "ISO-8859-9",
+        /* 11 DOS437 */ "CP437",
+        /* 12 DOS850 */ "CP850",
+        /* 13 DOS852 */ "CP852",
+        /* 14 DOS855 */ "CP855",
+        /* 15 DOS857 */ "CP857",
+        /* 16 DOS860 */ "CP860",
+        /* 17 DOS861 */ "CP861",
+        /* 18 DOS863 */ "CP863",
+        /* 19 DOS864 */ "CP864",
+        /* 20 DOS865 */ "CP865",
+        /* 21 DOS869 */ "CP869",
+        /* 22 DOS932 */ "CP932",
+        /* 23 MACINTOSH */ "MACINTOSH",
+        /* 24 BIG5 */ "BIG5",
+        /* 25 KSC5601 */ "CP949",
+        /* 26 JOHAB */ "JOHAB",
+        /* 27 DOS866 */ "CP866",
+        /* 28 ANSI_1250 */ "CP1250",
+        /* 29 ANSI_1251 */ "CP1251",
+        /* 30 ANSI_1252 */ "CP1252",
+        /* 31 GB2312 */ "GB2312",
+        /* 32 ANSI_1253 */ "CP1253",
+        /* 33 ANSI_1254 */ "CP1254",
+        /* 34 ANSI_1255 */ "CP1255",
+        /* 35 ANSI_1256 */ "CP1256",
+        /* 36 ANSI_1257 */ "CP1257",
+        /* 37 ANSI_874 */ "CP874",
+        /* 38 ANSI_932 */ "CP932",
+        /* 39 ANSI_936 */ "CP936",
+        /* 40 ANSI_949 */ "CP949",
+        /* 41 ANSI_950 */ "CP950",
+        /* 42 ANSI_1361 */ "CP1361",
+        /* 43 ANSI_1200 */ "UTF-16",
         /* 44 ANSI_1258 */ "CP1258"
     };
-    if( CADEncoding > 0 && CADEncoding < 45 )
+    if( CADEncoding > 0 && CADEncoding < 45 && CADEncoding != 4 )
```

Comment:
sizeof(pszSource) / sizeof(pszSource[0]) would be more robust than the hard coded 45